install/install.sh

```diff
@@ -45,328 +45,362 @@ error() {
+arch_check() {
+  arch=${arch:-"$(uname_arch)"}
+  case $arch in
+    x86_64) return 0 ;;
+  esac
+
+  platform=$(uname_platform)
+
+  error "$arch builds are not yet available for Starship"
+  printf "\n" >&2
+  info "If you would like to see a build for your configuration,"
+  info "please create an issue requesting a build for ${MAGENTA}${arch}-${platform}${NO_COLOR}:"
+  info "${BOLD}${UNDERLINE}https://github.com/starship/starship/issues/new/${NO_COLOR}"
+  return 1
+}
 
+bin_dir_check() {
   if [ ! -d "$BIN_DIR" ]; then
     error "Installation location $BIN_DIR does not appear to be a directory"
+    printf "\n" >&2
     info "Make sure the location exists and is a directory, then try again."
-    exit 1
+    return 1
   fi
 
-  # https://stackoverflow.com/a/11655875
-  local good
-  good=$(
-    IFS=:
-    for path in $PATH; do
-      if [ "${path}" = "${bin_dir}" ]; then
-        echo 1
-        break
-      fi
-    done
-  )
+  return 0
+}
 
-  if [ "${good}" != "1" ]; then
-    warn "Bin directory ${bin_dir} is not in your \$PATH"
-  fi
+path_check() {
+  # https://stackoverflow.com/a/9663359
+  case :$PATH: in
+    *:$BIN_DIR:*) return 0 ;;
+  esac
+
+  warn "Bin directory $BIN_DIR is not in your \$PATH"
+  printf "\n"
+
+  return 0
 }
 
-# defaults
-if [ -z "${PLATFORM-}" ]; then
-  PLATFORM="$(detect_platform)"
-fi
+confirm() {
+  printf "%s " "${MAGENTA}?${NO_COLOR} $* ${BOLD}[y/N]${NO_COLOR}"
+  {
+    read -r yn </dev/tty
+    rc=$?
+  } || true
 
-if [ -z "${BIN_DIR-}" ]; then
-  BIN_DIR=/usr/local/bin
-fi
+  if [ $rc != 0 ]; then
+    error "Error reading from prompt (please re-run with the \`--yes\` option)"
+    return 1
+  elif [ "$yn" != "y" ] && [ "$yn" != "yes" ]; then
+    error 'Aborting (please answer "yes" to continue)'
+    return 1
+  fi
 
-if [ -z "${ARCH-}" ]; then
-  ARCH="$(detect_arch)"
-fi
+  return 0
+}
 
-if [ -z "${BASE_URL-}" ]; then
-  BASE_URL="https://github.com/starship/starship/releases"
-fi
+# defaults
+PREFIX=${PREFIX:-"/usr/local"}
+BIN_DIR=${BIN_DIR:-"$PREFIX/bin"}
+
+elevated=
+force=
+verbose=
 
 # parse argv variables
 while [ "$#" -gt 0 ]; do
   case "$1" in
-  -p | --platform)
-    PLATFORM="$2"
-    shift 2
-    ;;
-  -b | --bin-dir)
-    BIN_DIR="$2"
-    shift 2
-    ;;
-  -a | --arch)
-    ARCH="$2"
-    shift 2
-    ;;
-  -B | --base-url)
-    BASE_URL="$2"
-    shift 2
-    ;;
-
-  -V | --verbose)
-    VERBOSE=1
-    shift 1
-    ;;
-  -f | -y | --force | --yes)
-    FORCE=1
-    shift 1
-    ;;
-
-  -p=* | --platform=*)
-    PLATFORM="${1#*=}"
-    shift 1
-    ;;
-  -b=* | --bin-dir=*)
-    BIN_DIR="${1#*=}"
-    shift 1
-    ;;
-  -a=* | --arch=*)
-    ARCH="${1#*=}"
-    shift 1
-    ;;
-  -B=* | --base-url=*)
-    BASE_URL="${1#*=}"
-    shift 1
-    ;;
-  -V=* | --verbose=*)
-    VERBOSE="${1#*=}"
-    shift 1
-    ;;
-  -f=* | -y=* | --force=* | --yes=*)
-    FORCE="${1#*=}"
-    shift 1
-    ;;
-
-  *)
-    error "Unknown option: $1"
-    exit 1
-    ;;
+    -p | --platform)
```

Comment:
Not opposed to capitalization changes, but curious if there's a rationale for them.

Reply:
I did it to emphasize the distinction between standard variables overridable from the environment (prefix, bin dir) and color modifiers vs other variables. Former are left uppercased while later got lowercased.

Reply:
Yeah, that seems like a good idea.